app.py

```diff
@@ -36,6 +26,7 @@ def inject_func():
 
 
 @app.route('/', methods=["GET"])
+@sql_pooling
```

Comment:
Having to put this directly on top of all handlers seems cumbersome. Why not use a Flask hook to do this automatically before every request?

See [`flask.Flask.before_request`](https://flask.palletsprojects.com/en/1.1.x/api/#flask.Flask.before_request) and this [SO question](https://stackoverflow.com/questions/22656403/python-flask-request-hook).